### ISSUE TITLE
Publish tab issues

### ISSUE THREAD

**Issue by abariska** [musescore/MuseScore#16709]
### Issue type

Other type of issue

### Bug description

As was discussed with @DmitryArefiev there shouldn't be an ability for users to change a score on Publish tab
There're issues (could be discussed):

- [ ] Transport panel can't be docked after was undocked. Probably, good to fix Transport panel on the main panel and don't let user to undock it.

https://user-images.githubusercontent.com/101250347/223748413-5b8e8d74-97ff-4d92-b7f8-0b29992809a8.mp4

- [ ] Playback can't be run by pressing Space button till you don't click Play button on the Transport panel.

https://user-images.githubusercontent.com/101250347/223749298-b7b96a76-7e38-445e-991b-98249b373a2d.mp4

- [ ] Score elements do not mark as selected when selected 

https://user-images.githubusercontent.com/101250347/223750610-d066e833-9788-44d6-8fea-6bdf45936029.mp4

- [ ] Publish Tab doesn't prevent changing pitches and dragging any score objects https://github.com/musescore/MuseScore/issues/13397 
- [ ] Note input isn’t disabled when switching to Publish tab https://github.com/musescore/MuseScore/issues/15277
- [ ] Inconsistency between view mode and view button state when switching between Score and Publish spaces https://github.com/musescore/MuseScore/issues/11800
- [ ] Undo and Redo buttons are available on the Publish tab

https://user-images.githubusercontent.com/101250347/223755942-8d388582-51e0-48a9-a2ff-f6ebdfbafa02.mp4

- [ ] Parts and Mixer buttons are enabled 

https://user-images.githubusercontent.com/101250347/223756378-f16ff7f7-8196-4cf7-888a-98107ee305c3.mp4

- [ ] Workspace, Concert pitch and View type are enabled
![abariska 2023-03-08 at 17 33 59](https://user-images.githubusercontent.com/101250347/223757617-5f26df6a-0ff9-4349-83d9-298f65248685.png)

### Steps to reproduce

See videos under each issue

### MuseScore Version

MuseScore 4.0.2 

### Regression

No.

### Operating system

macOS 12.6

**Comment by scorster** [comment]
*Transport panel can't be docked after was undocked. Probably, good to fix Transport panel on the main panel and don't let user to undock it.*

One concern. If Transport controls are permanently docked in the upper toolbar we lose the Tempo override control (and the flag buttons. 

The tempo override and flag buttons were features of MS3's Play Panel. But, both now and then, the flag buttons strike me as superfluous. It appears their only benefit is that users can add flags during playback—but the accuracy of such an endeavor is questionable. The most accurate way to define a Loop Length is by range selection. And it seems that should suffice. And perhaps there are other Flag button benefits not apparent to me.

Many people on MS3's forum miss the Play Panel, and Tempo override is clearly its most needed feature. (I was hoping to see that permanently in MS4's tool bar.)

On examining the MS3 Play Panel we see that most controls were redundancies—and I support that for creating a streamlined experience where all relevant controls appear in a single dialog. The only unique functions were the Tempo override slider (and field) and the flag buttons.

In short, please make sure the Tempo override slider and field are easily accessible and consider promoting them into the toolbar.

I'll start a request in the next few days for a revised and repurposed Play Panel, emphasizing redesigns that would dramatically elevate it's function as a practice/study guide and point to a number of forum discussions and MS3 Issue tracker posts.

scorster

**Comment by abariska** [comment]
@scorster This point (as rest of points in this issue) relates only to Publish tab, not Score. Nothing is going to be changed in Score tab relating to Transport panel.
And there was just my thoughts that all these playback features are really required on publish step.

**Comment by scorster** [comment]
Ha ha! Oops. Somehow I missed that your comments we specific to the Publish tab.

**Comment by RobFog** [comment]
> I'll start a request in the next few days for a revised/repurposed Play Panel, emphasizing redesigns that would dramatically elevate it's function as a practice/study guide and point to a number of forum discussions and MS3 Issue tracker posts.

I think this could this could still be useful.

**Comment by bkunda** [comment]
My thoughts on this:

1. It actually strikes me as odd that the playback panel can have different states for the "Score" and "Publish" tabs. Ideally, I'd like to see it retain whatever state the user has it in, regardless of which tab is in view. So, if the user undocks it in the Score tab, then it remains undocked (and in the same position) when switching over to the Publish tab. And then the user should be able to re-dock it in the publish tab, and it would remain re-docked when switching back over to the Score tab.

2. Agree that spacebar should be able to control playback in publish tab.

3. I actually don't mind the fact that score elements don't highlight when clicked. It pretty clearly communicates that the score is not editable in Publish mode. But it IS a bit odd that you can click on a note and hear it, but not receive any visual feedback to go with the interaction. Perhaps a more sophisticated interaction would be that **only** notes and rests could be clickable (to confirm the playback start point), but obviously they would not respond to any further interaction.

4. We should disable all the other components mentioned above (Undo, Redo, Parts, Mixer, Workspace, and Concert Pitch). Perhaps up for discussion though is whether Page View gets to remain independently configurable for the Score and Publish tabs. There might be an relevant use case for this: e.g. user is using the Publish tab as a quasi "presentation mode" (without the rest of the UI), and wants to present in Continuous view while being able to switch back to edit the score in Page view 🤷‍♂️

**Comment by MarcSabatella** [comment]
The "quasi presentation mode" would be my main use case for publish mode.  With that in mind, I'm in favor of any changes here that help with that, and biggest is probably maximizing available screen space for the score.

Eliminating the Parts and Mixer buttons would probably allow publish mode to get by with just one toolbar.  Eliminating the status controls would allow the status bar itself to be hidden.  In this usage, keyboard shortcuts are often the preferable way of doing things like zooming or toggling page/continuous view.  But, a well-design control that doesn't require much screen real estate would be nice - like a drop down menu on the single top toolbar.

For this purpose, I would also suggest that it should be as easy as possible to select a playback start point by clicking, since it would then be the one and only thing clicking is good for.  Any click anywhere in the score could be interested as a request to start playback there - and not just for a single staff.  Visual feedback need not show a note selected - it could simply display the vertical playback cursor.  Perhaps just briefly, so you can also have a clean view.

**Comment by abariska** [comment]
@bkunda Only thing I'd like to add to point 3:
Apart from notes and rests I'd leave also ability to select measures. As we see playback feature as an important one in Publish tab, selected measures could be helpful if user want to solo specific staff (as it's available in Score tab by default). Especially, if we take Publish mode as a quasi presentation mode

**Comment by bkunda** [comment]
Just on this point 3: a playback cursor might actually be the nicest solution here, as it then could eliminate any potentially misleading "selection" interaction (I.e. nothing in the score would be selectable, but anywhere you click in a measure would position a playback cursor, and playback would start from there).

Aside from this, I hadn't actually conceived of the publish tab to be used primarily as a presentation mode (although the thought of this use case had obviously occurred to me). It's cool that people might want to use it this way, but it wouldn't be it's only function; we still intend to add further publish/export functionality to it. In saying this, I think it obviously suggests the possibility of alternative use cases, which should be further investigated 🙂.

**Comment by abariska** [comment]
I'm not sure we understood each other correctly, so just will accent on my main thought about the point 3 once more 🙂
If we leave Playback panel same as in Score tab we should let user selecting measures. It's mandatory for Loop and (as I mentioned above) it will let soloing particular staff (instrument):

https://user-images.githubusercontent.com/101250347/225973994-3c8599b3-fc87-4820-8dd0-6a9ef610b5b4.mp4

**Comment by MarcSabatella** [comment]
I'm realizing that there are a number of other things that would really need to come together for the publish tab to really function well as a presentation mode.  The shortcuts for score navigation - zoom in/out, pgup/pgdn, etc - are also non-functional right now, for instance.

 Of course, someday maybe we could have "Present" as a fourth tab here, but without knowing more about the eventual direction of Publish, it's hard to say if there really is anything I'd want to do in Present that would interfere with the things one would do in Publish.  So if possible to keep going as if these use cases are compatible and only a single tab is needed, that would be preferable I think.

**Comment by scorster** [comment]
**@DmitryArefiev? wrote >** *...there shouldn't be an ability for users to change a score on Publish tab*

YES. So the question should be, "IF not, why not?" Or better yet, "If Publish view should provide editing, WHY?"

From the outset I've tried to wrap my head around the appearance of v4's grandiose Publish tab. Previously a "Publish" Toolbar item sufficed. And now all this? 

What value does the Publish view provide, particularly when Score view is one click away? Is the Publish tab supposed to preview things not evident/possible in Score view? If so, can someone supply a list of advantages to seeing/editing the score in Publish tab? Perhaps is the current Publish tab just the foundation for some larger not yet apparent concept?

The collateral damage is plenty of fuss and confusion about which "Score" features should be enabled/disabled in the Publish tab, and how to manage that AND how to keep the user from being confused.

**[abariska](https://github.com/abariska) wrote >** *If this ends up being really confusing for people* AND *But even this approach potentially introduces new possibilities for confusion, because there would be two kinds of selection behavior in two different parts of the app*

###As with many aspects of the v4 release, to Publish we must now go one level deeper:

• first we to the Publish tab, then to its Toolbar item "Publish to MuseScore.com, and we're rewarded with a stop where we can enter a Name, and have the options only to list as Public or Private ... Unlisted is available only when the user gets to .com's Score Edit "manage/upload" page. 

• And in MS3 one could tweak the settings there while the MP3 exported, now we must wait for the MP3 to export before getting to .com's  Score Edit.

**Marc wrote >** *The "quasi presentation mode" would be my main use case for publish mode. ... a number of other things that would need to come together for the publish tab to really function well as a presentation mode.* 

But why, when 100% presentation mode is right there in the Score tab? What am I missing?

Has anyone poled v4 users on their interactions and impressions of the Publish tab? Are they confused? Do they soundly benefit?

I'm failing to understand the advantages and goals in this design. Any insights?

scorster

**Comment by Tantacrul** [comment]
> What value does the v4's Publish tab provide, particularly when Score tab is one tab click away? Is the Publish tab supposed to preview things not evident/not possible in Score tab? If so, can anyone supply a list of advantages provided by seeing/editing/playing the score in Publish tab vs. Score tab? Perhaps the current Publish tab the foundation for some larger, not yet apparent concept?

I can help there. First, you need to remember that we don't implement full roadmaps in one release. We prepare the way for things we know we want to build and scale up our architecture accordingly. For example, we have the Home tab. This will eventually also contain demo scores, more plugins, stronger links to help documentation & a direct link to your score manager on MuseScore.com.

In the case of the publishing tab, we want to use it for the following:
- More granular settings when publishing to MuseScore.com
- Proper printing options that allow users more control over how their scores look and are combined
- Uploading to audio.com features
- and crucially... 'other stuff'. Essentially things we don't yet know we want to build

When designing V4, we made sure that the 'composing' part of the app was left alone so it didn't get ever more clunky and burdened with options. Hope this is helpful.

**Comment by scorster** [comment]
Thanks for the feedback. And apologies. I should have posted my comments in the Discussions area. 

scorster

**Comment by avvvvve** [comment]
Another related issue with in-score popups: #28937

**Comment by iainhallam** [comment]
I came to GitHub to report that pressing space on the Publish tab doesn't actually affect playback, though the tooltip for the play button still says that's the shortcut. I found that on this issue, but there seem to be many (and much larger) things mentioned here. Would it be worth splitting out some of the smaller parts of this issue into their own items that can be prioritised and tracked?

**Comment by cbjeukendrup** [comment]
@iainhallam That sounds like a good idea. And then we can link them as sub-issues to this one, to keep track of them.

**Comment by Jojo-Schmitz** [comment]
The mixer issue also needs a separate sub-issue (and came up in https://musescore.org/en/node/389453)
